components/compliance-service/reporting/util/xml.go

```diff
@@ -0,0 +1,133 @@
+package util
+
+import (
+	"encoding/xml"
+	"fmt"
+
+	reportingapi "github.com/chef/automate/api/interservice/compliance/reporting"
+)
+
+type xmlFields struct {
+	XMLName          xml.Name  `xml:"Node"`
+	ID               string    `xml:"ID,omitempty"`
+	NodeID           string    `xml:"NodeID,omitempty"`
+	NodeName         string    `xml:"NodeName,omitempty"`
+	EndTime          int64     `xml:"EndTime>Seconds,omitempty"`
+	Platform         Platform  `xml:"Platform,omitempty"`
+	Environment      string    `xml:"Environment,omitempty"`
+	IPAddress        string    `xml:"IPAddress,omitempty"`
+	FQDN             string    `xml:"FQDN,omitempty"`
+	Profiles         []Profile `xml:"Profiles>Profile,omitempty"`
+	Version          string    `xml:"Version,omitempty"`
+	ChefServer       string    `xml:"ChefServer,omitempty"`
+	ChefOrganization string    `xml:"ChefOrganization,omitempty"`
+}
+
+// Platform detail
+type Platform struct {
+	Name    string `xml:"Platform>Name,omitempty"`
+	Release string `xml:"Platform>Release,omitempty"`
+	Full    string `xml:"Platform>Full,omitempty"`
+}
+
+// Profile detail
+type Profile struct {
+	Name      string    `xml:"Name,omitempty"`
+	Title     string    `xml:"Title,omitempty"`
+	Version   string    `xml:"Version,omitempty"`
+	Summary   string    `xml:"Summary,omitempty"`
+	Controles []Control `xml:"Controles>Control,omitempty"`
+}
+
+// Control detail
+type Control struct {
+	ID                  string   `xml:"ID,omitempty"`
+	Title               string   `xml:"Title,omitempty"`
+	Impact              float32  `xml:"Impact,omitempty"`
+	Waived              string   `xml:"Waived,omitempty"`
+	WaiverJustification string   `xml:"Waiver>Justification,omitempty"`
+	WaiverExpiration    string   `xml:"Waiver>Expiration,omitempty"`
+	Results             []Result `xml:"Results>Result,omitempty"`
+}
+
+// Result detail
+type Result struct {
+	Status          string  `xml:"Status,omitempty"`
+	RunTime         float32 `xml:"RunTime,omitempty"`
+	CodeDescription string  `xml:"CodeDescription,omitempty"`
+	Message         string  `xml:"Message,omitempty"`
+	SkipMessage     string  `xml:"SkipMessage,omitempty"`
+}
+
+// ReportToXML converts a report to its XML representation as a string
+func ReportToXML(report *reportingapi.Report) ([]byte, error) {
```

Comment:
Would like to see some unit tests for this.

Reply:
nice job on this file!